libs/sdk-ui-filters/src/AttributeFilter/AttributeFilterButton/hooks/useAttributeFilterButtonState.ts

```diff
@@ -194,20 +195,17 @@ export const useAttributeFilterButtonState = (
         });
     };
 
-    const onSearch = useCallback(
-        debounce((query: string) => {
-            setState((s) => ({
-                ...s,
-                searchString: query,
-            }));
-        }, 500),
-        [],
-    );
+    const onSearch = debounce((query: string) => {
```

Comment:
This will not work properly, please use a ref for this, like here for example: https://github.com/gooddata/gooddata-ui-sdk/blob/master/libs/sdk-ui-dashboard/src/presentation/widget/kpi/KpiConfigurationPanel/KpiMetricDropdown/MetricDropdown.tsx#L42-L46
The reason for that is that you need to keep the same instance across renders, which only a ref can do.